src/taskgraph/util/dict_helpers.py

```diff
@@ -0,0 +1,10 @@
+def deep_get(dict_, field):
```

Comment:
I think this would be a good fit alongside the functions in `util/templates.py` rather than a new file.

Reply:
Also this can definitely be a follow-up, but there's similar logic for walking an object here:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/main.py#L96

Though in that case it's deleting rather than retrieving.